service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
+
+        Builder apiToken(BearerToken apiToken);
+
+        Builder security(SslConfiguration security);
```

Comment:
for immutable builders, you can omit the optional fields if they are absent.  I don't see the need to pass in a Optional typed value when you only need to set the value when it exists. 

Sure, lets use the old pattern if this is not convinient for other users. I will investigate/try to understand the problem more and update back with results.